### PR TITLE
Add more exception handling for geometry intersection

### PR DESCRIPTION
**Pull request by jwkaltz**
To be able to quickly identify the data which is problematic for Shapely V1.7 (see issue #1013 ), add exception handling and more logging.

### CODE REVIEW

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/lib/records/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=5406462&bid=17723306)

File: pyramid_oereb/lib/records/plr.py [hunk 1]
```diff
@@ -180,13 +180,18 @@ def calculate(self, real_estate):
...
+            try:
+                if geometry.calculate(
+                        real_estate,
+                        self.min_length, self.min_area,
+                        self.length_unit, self.area_unit
+                ):
+                    tested_geometries.append(geometry)
+                    inside = True
+            except Exception as ex:
+                log.info("PLR geometry check not possible, plr is {}, geometry.geom is {}"
```
Comment (kdeininger) [inline pyramid_oereb/lib/records/plr.py:192]: Why is this logged using `info()` instead of `error()`?

Comment (jwkaltz) [reply pyramid_oereb/lib/records/plr.py:192]: Thanks for having a look @kdeininger , in fact, I am not really happy with this PR, I was trying to output ids of the involved objects, but did not succeed; the real estate is clear, but I am not sure from where to get a meaningful key regarding the geometry being intersected with the real estate geometry, to make it easier to look it up in the data source. Maybe you have a better idea.

Comment (kdeininger) [reply pyramid_oereb/lib/records/plr.py:192]: You could try to calculate a point inside the invalid geometry:
https://shapely.readthedocs.io/en/maint-1.6/manual.html#object.representative_point

Using the point and the topic of the invalid PLR, it should be possible to identify the feature.

**Comment by vvmruder** [comment]
As mentioned in #1111 I think we not handle generally broken geometries in calculation. This should be done by simple (to be defined checks) on input into DB. PostGIS offers a lot of geometry health check functionality.

Indeed we should be able to give a more specific info about geometrically errors. Therefor I like @kdeininger suggestion. Instead of calculating a point in a broken geometry (which might fail too) we could simply use a point out of the geometry itself. Maybe a simple serialized linestring. This seems more robust to me.

**Pull request closed without merging**